File: LocalAIAgent.Domain/NewsArticle.cs
namespace LocalAIAgent.Domain
{
    public class NewsArticle
    {
        public required string Title { get; set; }
        public required string Summary { get; set; }
        public required DateTime PublishedDate { get; set; }
        public required string Link { get; set; }
        public required string Source { get; set; }
        public required HashSet<string> Categories { get; set; }
        public required Relevancy Relevancy { get; set; }
        public string? Reasoning { get; set; }
        public string? Topic { get; set; }
        public string? Event { get; set; }
        public int? InputTokens { get; set; }
        public int? OutputTokens { get; set; }
    }

    public enum Relevancy
    {
        High,
        Medium,
        Low
    }
}

File: LocalAIAgent.WebUI/src/clients/newsHub/NewsArticle.ts
import type { Relevancy } from './Relevancy';

export type NewsArticle = {
    title?: string | null;
    summary?: string | null;
    publishedDate: string;
    link?: string | null;
    source?: string | null;
    categories?: Array<string> | null;
    relevancy: Relevancy;
    reasoning?: string | null;
    topic?: string | null;
    event?: string | null;
    inputTokens?: number | null;
    outputTokens?: number | null;
};
